sphinx/environment/__init__.py

```diff
@@ -265,10 +273,10 @@ def __init__(self, app: Sphinx) -> None:
     def __getstate__(self) -> dict:
         """Obtains serializable data for pickling."""
         __dict__ = self.__dict__.copy()
-        __dict__.update(app=None, domains={}, events=None)  # clear unpickable attributes
-        # ensure that upon restoring the state, the most recent pickled files
-        # on the disk are used instead of those from a possibly outdated state
-        __dict__.update(_pickled_doctree_cache={})
+        # clear unpickable attributes
+        __dict__.update(app=None, domains={}, events=None)
+        # clear in-memory doctree caches
```

Comment:
I would like to keep the comment about the fact that by removing in-memory doctree caches, we use the files on the disk (at least someone who reads the comment understands that it's for that purpose, otherwise they might wonder "but why does it work anyway?)

Reply:
done